libafl/src/executors/inprocess.rs

```diff
@@ -483,16 +510,17 @@ mod unix_signal_handler {
             // Make sure we don't crash in the crash handler forever.
             data.current_input_ptr = ptr::null();
 
-            let interesting = state
+            let interesting = fuzzer
                 .objective_mut()
-                .is_interesting(&input, observers, &ExitKind::Crash)
+                .is_interesting(state, &input, observers, &ExitKind::Crash)
                 .expect("In crash handler objective failure.");
+
             if interesting {
                 let new_input = input.clone();
                 let mut new_testcase = Testcase::new(new_input);
-                state
+                fuzzer
                     .objective_mut()
-                    .append_metadata(&mut new_testcase)
+                    .append_metadata(state, &mut new_testcase)
```

Comment:
So why is Metadata no longer in State? Because of borrowing? That way, the state is no longer the complete state, though 🤔

Reply:
i don't get it. why it should not be in State? append_metadata insert a meta in the Testcase, than is then added to the corpus, that is in State.

Reply:
Ah sorry then it makes sense. I thought it's the fuzzer-wide metadata